### PR TITLE
ODS-5333 Postman Tests for Get Many Resources with Multiple Authorization Strategies 

### PR DESCRIPTION
**Pull request by semalaiappan**

### CODE REVIEW

File: Application/EdFi.Security.DataAccess/Repositories/SecurityRepository.cs [hunk 1]
```diff
@@ -32,6 +32,159 @@ public SecurityRepository(ISecurityContextFactory securityContextFactory)
...
+                                  ClaimSetResourceClaimAction = claimSetStudentSectionAssociationResourceClaimAction,
+                                  ClaimSetResourceClaimActionId = claimSetStudentSectionAssociationResourceClaimAction.ClaimSetResourceClaimActionId,
+                                  AuthorizationStrategy = authorizationStrategy,
+                                  AuthorizationStrategyId = authorizationStrategy.AuthorizationStrategyId
+                              });
+                        }
+                });
+
+                /*
+                var claimSetResourceClaimActionCreate = context.ClaimSetResourceClaimActions.Where(x => x.ResourceClaimId == studentSectionAssociationResourceClaimId && x.ClaimSetId == edFiSandboxClaimSetId
```
[the remark this answers is not in the document]
Comment (semalaiappan) [reply Application/EdFi.Security.DataAccess/Repositories/SecurityRepository.cs]: Fixed it. Thanks

File: Application/EdFi.Security.DataAccess/Repositories/SecurityRepository.cs [hunk 2]
```diff
@@ -32,6 +32,159 @@ public SecurityRepository(ISecurityContextFactory securityContextFactory)
...
+                              ClaimSetResourceClaimActionId = claimSetResourceClaimActionCreate.ClaimSetResourceClaimActionId,
+                              AuthorizationStrategy = authorizationStrategy,
+                              AuthorizationStrategyId = authorizationStrategy.AuthorizationStrategyId
+                          });
+                    }
+                });
+
+                */
+
+                var claimSetSeoraResourceClaimAction = context.ClaimSetResourceClaimActions.Where(x => x.ResourceClaimId == seoraResourceClaimId && x.ClaimSetId == edFiSandboxClaimSetId
```
[the remark this answers is not in the document]
Comment (semalaiappan) [reply Application/EdFi.Security.DataAccess/Repositories/SecurityRepository.cs]: Fixed it. Thanks

File: Application/EdFi.Security.DataAccess/Repositories/SecurityRepository.cs [hunk 3]
```diff
@@ -32,6 +32,159 @@ public SecurityRepository(ISecurityContextFactory securityContextFactory)
...
+                    && x.ActionId == createActionId).Single();
+
+
+                if (!context.ClaimSetResourceClaimActionAuthorizationStrategyOverrides.Any(a => a.ClaimSetResourceClaimActionId == claimSetSeoraResourceClaimAction.ClaimSetResourceClaimActionId
+                    && a.AuthorizationStrategyId == relationshipsWithEdOrgsOnlyAuthorizationStrategy.AuthorizationStrategyId))
+                    {
```
[the remark this answers is not in the document]
Comment (semalaiappan) [reply Application/EdFi.Security.DataAccess/Repositories/SecurityRepository.cs]: Fixed it. Thanks

File: Application/EdFi.Security.DataAccess/Repositories/SecurityRepository.cs [hunk 4]
```diff
@@ -32,6 +32,159 @@ public SecurityRepository(ISecurityContextFactory securityContextFactory)
...
+                }
+
+
+                if (claimSetResourceClaimActions.Any())
+                {
+                    context.ClaimSetResourceClaimActions.AddRange(claimSetResourceClaimActions);
+                    context.SaveChanges();
+                }
+
+                var claimSetStudentSectionAssociationResourceClaimAction = context.ClaimSetResourceClaimActions.Where(x => x.ResourceClaimId== studentSectionAssociationResourceClaimId && x.ClaimSetId == edFiSandboxClaimSetId
```
[the remark this answers is not in the document]
Comment (semalaiappan) [reply Application/EdFi.Security.DataAccess/Repositories/SecurityRepository.cs]: Fixed it. Thanks

File: Application/EdFi.Security.DataAccess/Repositories/SecurityRepository.cs [hunk 5]
```diff
@@ -32,6 +32,159 @@ public SecurityRepository(ISecurityContextFactory securityContextFactory)
...
+                var seoraResourceClaimId = resourceClaims.FirstOrDefault(x => x.ClaimName.Equals("http://ed-fi.org/ods/identity/claims/studentEducationOrganizationResponsibilityAssociation")).ResourceClaimId;
+
+                var readActionId = context.Actions.FirstOrDefault(a=>a.ActionName.ToLower()=="read").ActionId;
+                var createActionId = context.Actions.FirstOrDefault(a => a.ActionName.ToLower() == "create").ActionId;
+
+                var edFiSandboxClaimSetId = context.ClaimSets.FirstOrDefault(a => a.ClaimSetName == "Ed-Fi Sandbox").ClaimSetId;
+                var authorizationStrategyList = context.AuthorizationStrategies.ToList();
+                var authorizationStrategyReadList = authorizationStrategyList.Where(a => authorizationStrategyNameList.Contains(a.AuthorizationStrategyName)).ToList();
+                var relationshipsWithEdOrgsOnlyAuthorizationStrategy = authorizationStrategyList.Where(a => a.AuthorizationStrategyName.Equals("RelationshipsWithEdOrgsOnly")).Single();
```
[the remark this answers is not in the document]
Comment (semalaiappan) [reply Application/EdFi.Security.DataAccess/Repositories/SecurityRepository.cs]: Fixed it. Thanks

**Review by bptillman — APPROVED** [review]

**Pull request merged**